From 158525e21f8188a66ca847a1bd31823164cdf16e Mon Sep 17 00:00:00 2001
From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
Date: Fri, 27 May 2022 15:14:17 +0530
Subject: [PATCH] Enable independent SRIOV host setup from CIV setup

Changes:
- removed SRIOV-related setup functions
- added additional patching of qemu from SRIOV folder
- added prerequisite check for successfully completed SRIOV setup

Tracked-On:
Signed-off-by: Suresh, Prashanth <prashanth.suresh@intel.com>
---
 scripts/setup_host.sh | 86 ++++++++++++++-----------------------------
 1 file changed, 28 insertions(+), 58 deletions(-)

diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
index 59ae955..ffcedff 100755
--- a/scripts/setup_host.sh
+++ b/scripts/setup_host.sh
@@ -34,43 +34,6 @@ function ubu_changes_require(){
     sudo apt install -y libglib2.0-dev libncurses-dev libuuid1 uuid-dev libjson-c-dev
 }
 
-function ubu_install_sriov(){
-    # Switch to Xorg for Ubuntu 21.04
-    if [[ $(lsb_release -rs) == "21.04" ]]; then
-        sed -i "s/\#WaylandEnable=false/WaylandEnable=false/g" /etc/gdm3/custom.conf
-    fi
-
-    sudo apt install -y net-tools openssh-server git make autoconf libtool meson vim v4l-utils pkg-config libpciaccess-dev cmake \
-                        python3-pip python3.10 llvm-12-dev libelf-dev bison flex wayland-protocols libwayland-dev libwayland-egl-backend-dev \
-                        libx11-dev libxext-dev libxdamage-dev libx11-xcb-dev libxcb-glx0-dev libxcb-dri2-0-dev libxcb-dri3-dev \
-                        libxcb-present-dev libxshmfence-dev libxxf86vm-dev libxrandr-dev libkmod-dev libprocps-dev libdw-dev libpixman-1-dev \
-                        libcairo2-dev libudev-dev libgudev-1.0-0 gtk-doc-tools sshfs mesa-utils xutils-dev libunwind-dev libxml2-dev doxygen \
-                        xmlto cmake libpciaccess-dev graphviz libjpeg-dev libwebp-dev libsystemd-dev libdbus-glib-1-dev libpam0g-dev \
-                        freerdp2-dev libxkbcommon-dev libinput-dev libxcb-shm0-dev libxcb-xv0-dev libxcb-keysyms1-dev libxcb-randr0-dev \
-                        libxcb-composite0-dev libxcursor-dev liblcms2-dev libpango1.0-dev libglfw3-dev libxcb-composite0-dev libxcursor-dev \
-                        libgtk-3-dev libsdl2-dev virtinst virt-viewer virt-manager libspice-server-dev libusb-dev libxfont-dev libxkbfile-dev \
-                        libepoxy-dev rpm libncurses5-dev libncursesw5-dev liblz4-tool git-lfs uuid mtools python3-usb python3-pyudev \
-                        libjson-c-dev libfdt-dev socat bridge-utils uml-utilities python-dev libcap-ng-dev libusb-1.0-0-dev nasm acpidump \
-                        iasl libseccomp-dev libtasn1-6-dev libgnutls28-dev expect gawk
-    sudo apt install -y python3-mako
-
-    # Clean up any existing folders
-    function del_existing_folder() {
-        if [ -d "$1" ]; then
-            echo "Deleting existing folder $1"
-            rm -fr $1
-        fi
-    }
-    del_existing_folder $CIV_WORK_DIR/media
-    del_existing_folder $CIV_WORK_DIR/gstreamer
-    del_existing_folder $CIV_WORK_DIR/graphics
-    del_existing_folder $CIV_WORK_DIR/neo
-
-    # Start setup
-    source $CIV_WORK_DIR/scripts/sriov_setup_host.sh
-
-}
-
 function ubu_install_qemu_gvt(){
     sudo apt purge -y "^qemu"
     sudo apt autoremove -y
@@ -95,6 +58,14 @@ function ubu_install_qemu_gvt(){
         for i in $CIV_GOP_DIR/qemu/*.patch; do patch -p1 < $i; done
     fi
 
+    sriov_qemu_patch_num=$(ls $CIV_WORK_DIR/sriov_patches/qemu/*.patch 2> /dev/null | wc -l)
+    if [ "$sriov_qemu_patch_num" != "0" ]; then
+        for i in $CIV_WORK_DIR/sriov_patches/qemu/*.patch; do
+            echo "applying qemu patch $i"
+            patch -p1 < $i
+        done
+    fi
+
     vertical_qemu_patch_num=$(ls $CIV_VERTICAl_DIR/qemu/*.patch 2> /dev/null | wc -l)
     if [ "$vertical_qemu_patch_num" != "0" ]; then
         for i in $CIV_VERTICAl_DIR/qemu/*.patch; do
@@ -213,25 +184,6 @@ function ubu_enable_host_gvt(){
     fi
 }
 
-function ubu_enable_host_sriov(){
-    if [[ ! `cat /etc/default/grub` =~ "i915.enable_guc=0x7 udmabuf.list_limit=8192" ]]; then
-        read -p "The grub entry in '/etc/default/grub' will be updated for enabling SRIOV, do you want to continue?     [Y/n]" res
-        if [ x$res = xn ]; then
-            return
-        fi
-        if [[ ! `cat /etc/default/grub` =~ "intel_iommu=on i915.force_probe=*" ]]; then
-            sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"intel_iommu=on i915.force_probe=* /g" /etc/default/grub
-        fi
-        sed -i "s/GRUB_CMDLINE_LINUX=\"/GRUB_CMDLINE_LINUX=\"i915.enable_guc=0x7 udmabuf.list_limit=8192 /g" /etc/default/grub
-        update-grub
-
-        echo -e "\nkvmgt\nvfio-iommu-type1\nvfio-mdev\n" >> /etc/initramfs-tools/modules
-        update-initramfs -u -k all
-
-        reboot_required=1
-    fi
-}
-
 function ubu_update_fw(){
     FW_REL="linux-firmware-20220310"
     GUC_REL="70.0.3"
@@ -286,6 +238,25 @@ function check_kernel_version() {
     fi
 }
 
+function check_sriov_setup() {
+    input="$CIV_WORK_DIR/sriov_setup_ubuntu.log"
+    sriov_setup_success=0
+
+    if [ -f "$input" ]; then
+        while read -r line
+        do
+            if [[ $line == "Success" ]]; then
+                sriov_setup_success=1
+            fi
+        done < "$input"
+    fi
+
+    if [ $sriov_setup_success == 0 ]; then
+        echo "E: Please ensure SRIOV has been setup successfully first"
+        exit
+    fi
+}
+
 function ask_reboot(){
     if [ $reboot_required -eq 1 ];then
         read -p "Reboot is required, do you want to reboot it NOW? [y/N]" res
@@ -581,13 +552,12 @@ parse_arg "$@"
 check_os
 check_network
 check_kernel_version
+check_sriov_setup
 
 ubu_changes_require
-ubu_install_sriov
 ubu_install_qemu_gvt
 ubu_build_ovmf_gvt
 ubu_enable_host_gvt
-ubu_enable_host_sriov
 ubu_update_fw
 
 install_vm_manager
-- 
2.36.1

